logging.go

```diff
@@ -1,35 +1,45 @@
 package sgreplicate
 
 import (
-	"fmt"
-	"strings"
-
 	"github.com/couchbase/clog"
 )
 
-type LoggingReplication struct {
-	Parameters ReplicationParameters
+const logKey = "Replicate"
+
+func (r *Replication) log(level clog.LogLevel, format string, args ...interface{}) {
+	log(r.Parameters, level, format, args)
 }
 
-func (lr LoggingReplication) LogTo(key string, format string, args ...interface{}) {
-	clog.To(key, prefixWithReplicationId(lr, format), args...)
+func (r *ContinuousReplication) log(level clog.LogLevel, format string, args ...interface{}) {
+	log(r.Parameters, level, format, args)
 }
 
-func (lr LoggingReplication) Warn(args ...interface{}) {
-	if clog.Level <= clog.LevelWarning {
-		clog.Warn(prefixWithReplicationId(lr, fmt.Sprint(args...)))
+func log(param ReplicationParameters, level clog.LogLevel, format string, args ...interface{}) {
+	if param.LogFn == nil {
```

Comment:
Doing this check on each invocation of log seems a bit much?

Can we not rely on setting the default in `NewReplication()`?

Reply:
This doesn't work in the tests as the 'constructor' isn't always ran first. This check was for test logging... Perhaps there may be another way around.

Reply:
I guess moving the work to the test side may be better my TestMain seems to be too much for it?